CMakeLists.txt

```diff
@@ -203,7 +203,6 @@ if("${isSystemDir}" STREQUAL "-1")
   file(RELATIVE_PATH BINTOPREFIXRELPATH ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR} ${CMAKE_INSTALL_PREFIX})
   file(RELATIVE_PATH PREFIXTOLIBRELPATH ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
   if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    set(CMAKE_INSTALL_NAME_DIR "@rpath/${PREFIXTOLIBRELPATH}")
```

Comment:
We should definitely be setting the install name for our libraries. Why did this get removed?

Reply:
I've been looking into the CMake docs more. Note we do `set(CMAKE_MACOSX_RPATH true)`, which means that, per https://cmake.org/cmake/help/latest/prop_tgt/MACOSX_RPATH.html#prop_tgt:MACOSX_RPATH, 

> the directory portion of the install_name field of this shared library will be `@rpath` unless overridden by [INSTALL_NAME_DIR](https://cmake.org/cmake/help/latest/prop_tgt/INSTALL_NAME_DIR.html#prop_tgt:INSTALL_NAME_DIR). This indicates the shared library is to be found at runtime using runtime paths (rpaths).

Thus the install_name is getting set regardless, and the question is whether we want it to default to `@rpath` or have it be `@rpath/lib`

Note also I think setting `CMAKE_MACOSX_RPATH` is redundant based on the version of CMake we're using:
https://cmake.org/cmake/help/latest/policy/CMP0042.html#policy:CMP0042